webserver/src/layouts/Header.tsx

```diff
@@ -1,26 +1,35 @@
 import { lazy, Suspense } from 'react'
 
-import SNFG from '../components/PrivateerDisplay/SNFG'
+import SNFG from '../components/PrivateerDisplay/SNFG.tsx'
 
-const Upload = lazy(() => import('../components/Upload/Upload'));
-const Loading = lazy(() => import('../components/Loading/Loading'));
-const NavBar = lazy(() => import('../layouts/NavBar'));
+const Upload = lazy(() => import('../components/Upload/Upload.tsx'));
+const Loading = lazy(() => import('../components/Loading/Loading.tsx'));
+const NavBar = lazy(() => import('./NavBar.tsx'));
+const NoGlycans = lazy(() => import("../components/NoGlycans/NoGlycans.tsx"))
 
-const NoGlycans = lazy(() => import("../components/NoGlycans/NoGlycans"))
+import {HeaderProps} from "../interfaces/types"
 
-export function Header(props) {
+export function Header(props: HeaderProps) {
+    
+    let filename = ""
+    if (props.PDBCode != "") { 
+        filename = props.PDBCode
+    }
+    else if (props.coordinateFile) { 
+        filename = props.coordinateFile.name
+    }
 
     return (
         <div className="bg-gray text-primary">
             <NavBar setResetApp={props.setResetApp}/>
             <div className="flex justify-center mb-6">
                 {props.fallback != true ?
-                    <Suspense fallback={<Loading loadingText={"Loading content..."} />}>
-                        {props.submit == null ?
+                    <Suspense fallback={<Loading loadingText={"Loading"} />}>
+                        {props.submit == false ?
```

Comment:
I think this is a javascript typo. I think you want to have  `===` instead of `==`?